### PR TITLE
Dense to sparse matrix

### LINKED ISSUE #274: Convert dense matrix to sparse matrix
This would speed up my matrix multiplies.

### PR DESCRIPTION
**Pull request by ivuckovic**
#### Reference Issue
Closes #274 

#### What does your implementation fix? 
Implements dense to sparse matrix conversion

#### Explain your changes.
Implement sparse function in math.py
   - implemented function uses scipy sompressed sparse row matrix format to create sparse matrix
   - function returns TensorBase object
Implement spars functions in tensor.py
  - sparse() returns TensorBase object same as sparse() in math.py
  - sparse_() operates inline
Add unit tests for both sparse() functions and for sparse_()

### CODE REVIEW

File: syft/math.py [hunk 1]
```diff
@@ -845,3 +849,24 @@ def renorm(tensor1, p, dim, maxnorm):
...
         out = tensor1 * scalar_reshaped
     return TensorBase(out)
+
+def sparse(tensor):
+    """
+    Converts dense matrix to sparse, returning a new matrix as a tensor
+
+    Parameters
+    ----------
+    tensor1: TensorBase
```
Comment (bharathgs) [inline syft/math.py]: it should be `tensor`.

File: syft/tensor.py [hunk 2]
```diff
@@ -3367,6 +3367,43 @@ def stride(self, dim=None):
...
+
+        return syft.math.sparse(self)
+
+    def sparse_(self):
+        """
+		Converts dense matrix to sparse, returning a new matrix as a tensor
+
+		Parameters
+		----------
+		tensor1: TensorBase
```
Comment (bharathgs) [inline syft/tensor.py]: this as well (`tensor`)

**Comment by bharathgs** [comment]
@ivuckovic  could you please rewrite these as per #350

**Comment by iamtrask** [comment]
seems to be hanging

**Comment by ivuckovic** [comment]
@bharathgs, thanks for review. I'll get on to it right away.

**Pull request closed without merging**